File: subroutines/call-stack/java/calculate_product.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Scanner;

class CallStack
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        // Use Scanner for reading user input from the console
        Scanner scanner = new Scanner(System.in);

        // Demonstrates the use of call stacks
        System.out.println("Please enter a number");
        String num1String = scanner.nextLine();
        int num1 = Integer.parseInt(num1String);

        System.out.println("Please enter another number");
        String num2String = scanner.nextLine();
        int num2 = Integer.parseInt(num2String);

        calculateProduct(num1, num2);
    }


    // Calculates the product of two numbers
    public static void calculateProduct(int n1, int n2)
    {
        int product = n1 * n2;
        System.out.println(product);
    }


}

File: subroutines/call-stack/python/calculate_product.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def calculate_product(n1, n2):
    """Calculates the product of two numbers"""
    product = n1 * n2
    print(product)


def main():
    """Simple program to demonstrate use of call stack"""
    num1_input = input("Enter a number ")
    num2_input = input("Enter another number ")
    num1 = int(num1_input)
    num2 = int(num2_input)
    calculate_product(num1, num2)
    print("I hope you enjoyed using the product calculator")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()
